tests/ext/pynamodb/test_pynamodb.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from pynamodb.attributes import UnicodeAttribute
+from pynamodb.exceptions import VerboseClientError
+from pynamodb.models import Model
+
+from aws_xray_sdk.core import patch
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+
+patch(('pynamodb',))
+
+
+@pytest.fixture(autouse=True)
+def construct_ctx():
+    """
+    Clean up context storage on each test run and begin a segment
+    so that later subsegment can be attached. After each test run
+    it cleans up context storage again.
+    """
+    xray_recorder.configure(service='test', sampling=False, context=Context())
+    xray_recorder.clear_trace_entities()
+    xray_recorder.begin_segment('name')
+    yield
+    xray_recorder.clear_trace_entities()
+
+
+def test_exception():
+    class SampleModel(Model):
+        class Meta:
+            region = 'us-west-2'
+            table_name = 'mytable'
+
+        sample_attribute = UnicodeAttribute(hash_key=True)
+
+    try:
+        SampleModel.describe_table()
+    except VerboseClientError:
+        pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.error
+    assert subsegment.http['response']['status'] == 400
```

Comment:
What exception are you trying to produce? Is the code always guaranteed to be 400 or it could be different depend on users and machines?

Reply:
These tests currently do real requests to DynamoDB, as I don't know how to avoid them. Using the botocore stubber, as in the botocore tests, doesn't work, because we're patching requests here directly. The requests made by the tests currently return a access denied unless there are is access to resources called "mytable" and "mybucket" available.